validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/HttpClientExternalSignerFactory.java

```diff
@@ -58,9 +68,22 @@ public HttpClient get() {
     return builder.build();
   }
 
-  private boolean isTLSEnabled() {
-    final String protocol = validatorConfig.getValidatorExternalSignerUrl().getProtocol();
-    return protocol != null && protocol.equalsIgnoreCase("https");
+  private boolean isTLSEnabled(final URL externalSignerUrl) {
+    return "https".equalsIgnoreCase(externalSignerUrl.getProtocol());
+  }
+
+  private void configureBasicAuthentication(
+      final HttpClient.Builder builder, final String userInfo) {
+    final List<String> authCredentials = Splitter.on(':').splitToList(userInfo);
+    final String username = authCredentials.get(0);
```

Comment:
This breaks if userInfo doesn't contain `:` you should check authCredentials size

Reply:
I am not sure about that. Isn't that standard to use the `user:password` format

Reply:
But you can always pass an url without ':' and seems like URL constructor doesn't complain. (I tried in test)

Reply:
Yeah, get it. Added an exception for that case.